File: src/js/factorial_float.js
const { performance } = require('perf_hooks');
const count = parseInt(process.env.COUNT) || 170;

let factorial = 1.0;

const start = performance.now();
for (let i = 1; i <= count; i++) {
    factorial *= i;
}
const end = performance.now();

console.log(`Result(${count}!): ${factorial.toExponential(10)}`);
console.log(`Time: ${(end - start).toFixed(3)} ms`);

File: src/rust/factorial_float.rs
use std::env;
use std::time::Instant;

fn main() {
    let count_str = env::var("COUNT").unwrap_or("170".to_string());
    let count = count_str.parse::<u64>().unwrap_or(170);

    let mut factorial = 1.0f64;
    
    let start = Instant::now();
    for i in 1..=count {
        factorial *= i as f64;
    }
    let duration = start.elapsed();

    println!("Result({}!): {:e}", count, factorial);
    println!("Time: {:.3} ms", duration.as_nanos() as f64 / 1_000_000.0);
}
